actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -31,16 +31,23 @@ class InvalidCrossOriginRequest < ActionControllerError #:nodoc:
   # URL on your site. When your JavaScript response loads on their site, it executes.
   # With carefully crafted JavaScript on their end, sensitive data in your JavaScript
   # response may be extracted. To prevent this, only XmlHttpRequest (known as XHR or
-  # Ajax) requests are allowed to make GET requests for JavaScript responses.
+  # Ajax) requests are allowed to make requests for JavaScript responses.
   #
-  # It's important to remember that XML or JSON requests are also affected and if
-  # you're building an API you should change forgery protection method in
+  # It's important to remember that XML or JSON requests are also checked by default. If
+  # you're building an API or an SPA you could change forgery protection method in
   # <tt>ApplicationController</tt> (by default: <tt>:exception</tt>):
   #
   #   class ApplicationController < ActionController::Base
   #     protect_from_forgery unless: -> { request.format.json? }
```

Comment:
Because this applies generally to XHR requests, should this guidance also be changed to `request.xhr?` instead of `request.format.json?`.

Reply:
It's just an example, but sounds like a good idea to cover the general case 👍 (happy to change it, but I didn't get any feedback on this PR so far and not sure if anybody's interested unfortunately)

Reply:
I happened to find your PR while researching the safety of excluding XHRs from CSRF protection. The clarifications you made to the documentation here was very helpful to me, for what it's worth.

Given that it appears to be generally safe to do so (I found a more detailed explanation [here](https://markitzeroday.com/x-requested-with/cors/2017/06/29/csrf-mitigation-for-ajax-requests.html)), I'm also wondering whether it instead makes sense for Rails to exclude CSRF protection for XHR calls by default (i.e., should `verified_request?` simply return true for `request.xhr?` too?): https://github.com/rails/rails/blob/0a6b866ff2f45cea50598fc054763b89ea7affbe/actionpack/lib/action_controller/metal/request_forgery_protection.rb#L283-L291

With this change, you would no longer need to add the boilerplate `unless: -> { request.xhr? }` to every API-driven Rails project.

Reply:
Thanks for the positive feedback. Those suggested changes to the doc came about after a little of my own frustration trying to understand whether it's safe or not, or in which situations exactly. 

Please take a look at the second part of the suggested updates however. Whilst generally safe, there is a subtle risk here if combined with too permissive CORS I believe. I too often see places that define wildcard `Access-Control-Allow-Origin "*"`. Usually as a quickfix to solving cross-domain limitations. When combined, sites open themselves to CSRF again as far as I can tell.

So I would still imagine this is a safer / more sane default to have. But of course if you know what you're doing, and don't open up CORS to everyone, it seems totally safe to bypass CSRF protection to XHR.

I hope this PR can be merged so that the advice is clearer. I'm not entirely sure what needs to happen for it to get merged though. I'm not a regular contributor to rails. It's my first PR.